### PR TITLE
feat: add legal information endpoints

### PR DESCRIPTION
**Pull request by dajay**

### CODE REVIEW

File: src/main/java/fr/redfroggy/ilg/client/legalinformation/LegalStatus.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  public String getRegistrationId() {
+    return registrationId;
+  }
+
+  /**
+   * Deposit date
+   * @return date
+  */
+  @ApiModelProperty(value = "Deposit date")
+  public LocalDate getDate() {
```
Comment (jgavignet) [inline src/main/java/fr/redfroggy/ilg/client/legalinformation/LegalStatus.java:46]: @Valid ?

File: src/main/java/fr/redfroggy/ilg/client/legalinformation/LegalStatus.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    return date;
+  }
+
+  /**
+   * Act date
+   * @return dateAct
+  */
+  @ApiModelProperty(value = "Act date")
+  public LocalDate getDateAct() {
```
Comment (jgavignet) [inline src/main/java/fr/redfroggy/ilg/client/legalinformation/LegalStatus.java:56]: @Valid ?

File: src/main/java/fr/redfroggy/ilg/client/legalinformation/LegalAct.java [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+    }
+
+    public LegalActBuilder available(Boolean available) {
+      this.available = available;
+      return this;
+    }
+
+    public LegalAct build() {
+      LegalAct legalAct = new LegalAct(this);
+      return legalAct;
```
Comment (jgavignet) [inline src/main/java/fr/redfroggy/ilg/client/legalinformation/LegalAct.java]: return new LegalAct(this);

**Review by jgavignet — APPROVED** [review]

File: src/main/java/fr/redfroggy/ilg/client/legalinformation/LegalDeposit.java [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+  @ApiModelProperty(example = "456", value = "Deposit number")
+  public Integer getNumber() {
+    return number;
+  }
+
+  /**
+   * where the deposit happened
+   * @return date
+  */
+  @ApiModelProperty(value = "where the deposit happened")
```
Comment (michaeldesigaud) [inline src/main/java/fr/redfroggy/ilg/client/legalinformation/LegalDeposit.java]:
```suggestion
  @ApiModelProperty(value = "when the deposit happens")
```

File: src/main/java/fr/redfroggy/ilg/client/legalinformation/LegalDeposit.java [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+   * Deposit number
+   * @return number
+  */
+  @ApiModelProperty(example = "456", value = "Deposit number")
+  public Integer getNumber() {
+    return number;
+  }
+
+  /**
+   * where the deposit happened
```
Comment (michaeldesigaud) [inline src/main/java/fr/redfroggy/ilg/client/legalinformation/LegalDeposit.java]:
```suggestion
   * when the deposit happens
```

File: src/main/java/fr/redfroggy/ilg/client/legalinformation/LegalStatus.java (hunk 2 above)
Comment (dajay) [reply src/main/java/fr/redfroggy/ilg/client/legalinformation/LegalStatus.java:56]: done

**Review by michaeldesigaud — APPROVED** [review]

File: src/main/java/fr/redfroggy/ilg/client/legalinformation/LegalDeposit.java (hunk 4 above)
Comment (dajay) [reply src/main/java/fr/redfroggy/ilg/client/legalinformation/LegalDeposit.java]: done

File: src/main/java/fr/redfroggy/ilg/client/legalinformation/LegalDeposit.java (hunk 5 above)
Comment (dajay) [reply src/main/java/fr/redfroggy/ilg/client/legalinformation/LegalDeposit.java]: done

**Pull request merged**